pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -326,6 +331,74 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 				assert.Nil(t, uninstallJob)
 			},
 		},
+		{
+			name: "Resolve installer image from spec.images.installerimage",
+			existing: []runtime.Object{
+				func() *hivev1.ClusterDeployment {
+					cd := testClusterDeployment()
+					cd.Status.InstallerImage = nil
+					cd.Spec.Images.InstallerImage = "test-installer-image:latest"
+					return cd
+				}(),
+				testSecret(adminPasswordSecret, adminCredsSecretPasswordKey, "password"),
+				testSecret(pullSecretSecret, pullSecretKey, "{}"),
+				testSecret(sshKeySecret, adminSSHKeySecretKey, "fakesshkey"),
+			},
+			validate: func(c client.Client, t *testing.T) {
+				cd := getCD(c)
+				if cd.Status.InstallerImage == nil || *cd.Status.InstallerImage != "test-installer-image:latest" {
+					t.Errorf("unexpected status.installerImage")
+				}
+			},
+		},
+		{
+			name: "Resolve installer image from imageSet.spec.installerimage",
+			existing: []runtime.Object{
+				func() *hivev1.ClusterDeployment {
+					cd := testClusterDeployment()
+					cd.Status.InstallerImage = nil
+					cd.Spec.Images.InstallerImage = ""
+					cd.Spec.ImageSet = &hivev1.ClusterImageSetReference{Name: testClusterImageSetName}
+					return cd
+				}(),
+				func() *hivev1.ClusterImageSet {
+					cis := testClusterImageSet()
+					cis.Spec.InstallerImage = strPtr("test-cis-installer-image:latest")
+					return cis
+				}(),
+				testSecret(adminPasswordSecret, adminCredsSecretPasswordKey, "password"),
+				testSecret(pullSecretSecret, pullSecretKey, "{}"),
+				testSecret(sshKeySecret, adminSSHKeySecretKey, "fakesshkey"),
+			},
+			validate: func(c client.Client, t *testing.T) {
+				cd := getCD(c)
+				if cd.Status.InstallerImage == nil || *cd.Status.InstallerImage != "test-cis-installer-image:latest" {
+					t.Errorf("unexpected status.installerImage")
+				}
+			},
+		},
+		{
+			name: "Create job to resolve installer image",
+			existing: []runtime.Object{
+				func() *hivev1.ClusterDeployment {
+					cd := testClusterDeployment()
+					cd.Status.InstallerImage = nil
+					cd.Spec.Images.InstallerImage = ""
+					cd.Spec.ImageSet = &hivev1.ClusterImageSetReference{Name: testClusterImageSetName}
+					return cd
+				}(),
+				testClusterImageSet(),
+				testSecret(adminPasswordSecret, adminCredsSecretPasswordKey, "password"),
+				testSecret(pullSecretSecret, pullSecretKey, "{}"),
+				testSecret(sshKeySecret, adminSSHKeySecretKey, "fakesshkey"),
+			},
+			validate: func(c client.Client, t *testing.T) {
+				job := getImageSetJob(c)
+				if job == nil {
+					t.Errorf("did not find expected imageset job")
+				}
+			},
+		},
```

Comment:
Could you add a test for deprovision, when the CD references a ClusterImageSet that has since been deleted. I'm not quite sure what we want there but I think it might be ok, we don't need an installer image anymore, if we do start using the installer image to deprovision in future we still have a reference at least to what was used to install the cluster, although that's probably no good anymore. For now though, we'd just use the latest hive image defaulted in the code, which I think is correct as we want it to remain backwards compat.